ui/UIElement.lua

```diff
@@ -0,0 +1,84 @@
+local class = require("class")
+
+local uniqueId = 0
+
+--@module Button
+local UIElement = class(
+  function(self, options)
+    self.id = uniqueId
+    uniqueId = uniqueId + 1
+    
+    self.x = options.x or 0
+    self.y = options.y or 0
+    self.width = options.width or 110
+    self.height = options.height or 25
+    self.label = options.label or ""
+    self.extraLabels = options.extra_labels or {}
+    self.translate = options.translate or options.translate == nil and true

```

Comment:
Is there a reason for the or/and for 17, 18, and 19?
```suggestion
    self.translate = options.translate
```

Reply:
these fields should have a default value of true, without that they will have a default value of nil and will force you to explicitly set all elements as visible/enabled/translate on creation.